packages/remix-dev/vite/plugin.ts

```diff
@@ -416,19 +419,17 @@ const getServerBundleBuildConfig = (
   return viteUserConfig.__remixServerBundleBuildConfig as ServerBundleBuildConfig;
 };
 
-let getServerBuildDirectory = (
-  remixConfig: ResolvedVitePluginConfig,
-  { serverBundleId }: Pick<BuildContext, "serverBundleId">
-) =>
+let getServerBuildDirectory = (buildContext: BuildContext) =>
   path.join(
-    remixConfig.buildDirectory,
+    buildContext.remixConfig.buildDirectory,
     "server",
-    ...(typeof serverBundleId === "string" ? [serverBundleId] : [])
+    ...(typeof buildContext.serverBundleId === "string"
+      ? [buildContext.serverBundleId]
+      : [])
   );
 
-export let getServerBuildRootDirectory = (
-  remixConfig: ResolvedVitePluginConfig
-) => getServerBuildDirectory(remixConfig, { serverBundleId: undefined });
+export let getServerBuildRootDirectory = (buildContext: BuildContext) =>
+  getServerBuildDirectory(buildContext);
```

Comment:
```suggestion
export let getServerBuildRootDirectory = getServerBuildDirectory;
```

Though at that point, might be worth just exporting `getServerBuildDirectory` directly

Reply:
Nice catch, I missed this!